### ISSUE TITLE
S3 open fails on files that contain '@' in their path

### ISSUE THREAD

**Issue by mayk0gan** [piskvorky/smart_open#779]
#### Problem description

Be sure your description clearly answers the following questions:

- What are you trying to achieve?
- What is the expected result?
- What are you seeing instead?

#### Steps/code to reproduce the problem

In order for us to be able to solve your problem, we have to be able to reproduce it on our end.
Without reproducing the problem, it is unlikely that we'll be able to help you.

Include full tracebacks, logs and datasets if necessary.
Please keep the examples minimal ([minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)).

**When trying to use `smart_open.open` on a S3 file that contains '@' in its name, I get the following exception:**

```
 with smart_open.open(
 File \"/usr/lib/python3.8/site-packages/smart_open/smart_open_lib.py\", line 224, in open
 binary = _open_binary_stream(uri, binary_mode, transport_params)
 File \"/usr/lib/python3.8/site-packages/smart_open/smart_open_lib.py\", line 400, in _open_binary_stream
 fobj = submodule.open_uri(uri, mode, transport_params)
 File \"/usr/lib/python3.8/site-packages/smart_open/s3.py\", line 221, in open_uri
 parsed_uri = parse_uri(uri)
 File \"/usr/lib/python3.8/site-packages/smart_open/s3.py\", line 110, in parse_uri
 access_id, access_secret = auth.split(':')
ValueError: too many values to unpack (expected 2)
```

You can reproduce this scenario using the following code:

```python
from smart_open.s3 import parse_uri

url = "s3://bucket-name/a12b34cd-5e67-890f-1234-56d789012e34/Re: Subject @ Tue 18 Sept 2019 11:15am - 11:30am (john@example.com).eml"
parse_uri(url)
```

Output:

```
File ..../py3.8/lib/python3.8/site-packages/smart_open/s3.py:110, in parse_uri(uri_as_string)
    108 if '@' in uri and ':' in uri.split('@')[0]:
    109     auth, uri = uri.split('@', 1)
--> 110     access_id, access_secret = auth.split(':')
    112 head, key_id = uri.split('/', 1)
    113 if '@' in head and ':' in head:

ValueError: too many values to unpack (expected 2)
```

#### Versions

Please provide the output of:

```python
import platform, sys, smart_open
print(platform.platform())
print("Python", sys.version)
print("smart_open", smart_open.__version__)
```

```
macOS-12.4-arm64-arm-64bit
Python 3.8.16 (default, Dec  7 2022, 01:24:57)
[Clang 14.0.0 (clang-1400.0.29.202)]
smart_open 6.3.0
```

#### Checklist

Before you create the issue, please make sure you have:

- [X] Described the problem clearly
- [X] Provided a minimal reproducible example, including any required data
- [X] Provided the version numbers of the relevant software

**Comment by mayk0gan** [comment]
seems related https://github.com/RaRe-Technologies/smart_open/pull/776

**Comment by rileypeterson** [comment]
Seems to work fine after merge of #776 👍 :
```
>>> from smart_open.s3 import parse_uri
>>> url = "s3://bucket-name/a12b34cd-5e67-890f-1234-56d789012e34/Re: Subject @ Tue 18 Sept 2019 11:15am - 11:30am (john@example.com).eml"
>>> parse_uri(url)
{'scheme': 's3', 'bucket_id': 'bucket-name', 'key_id': 'a12b34cd-5e67-890f-1234-56d789012e34/Re: Subject @ Tue 18 Sept 2019 11:15am - 11:30am (john@example.com).eml', 'port': 443, 'host': 's3.amazonaws.com', 'ordinary_calling_format': False, 'access_id': None, 'access_secret': None}
```

**Comment by mayk0gan** [comment]
@rileypeterson, that's amazing! Could you please let me know when the next release is scheduled? Thank you!

**Comment by rileypeterson** [comment]
@mayk0gan I don't know when the next release is scheduled

**Comment by ddelange** [comment]
released in v6.4.0
